packages/gatsby-plugin-image/src/components/main-image.tsx

```diff
@@ -5,7 +5,26 @@ export type MainImageProps = PictureProps
 
 export const MainImage = forwardRef<HTMLImageElement, MainImageProps>(
   function MainImage({ ...props }, ref) {
-    return <Picture ref={ref} {...props} />
+    return (
+      <>
+        {props.loading === `eager` && (
+          <link
```

Comment:
does this actually speed things up? preload helps by a) upping the loading priority within the browser and b) letting the browser discover the resource earlier than it would otherwise.

Since images are already loaded at the highest priority & this is rendered right before the image — it wouldn't change much perhaps?

Might be wrong somewhere in there. But curious if you tested and saw a diff?